cli/src/semgrep/app/scans.py

```diff
@@ -338,6 +339,21 @@ def report_findings(
 
         dependency_counts = {k: len(v) for k, v in lockfile_dependencies.items()}
 
+        # NOTE: This mirrors the logic in metrics.py to show the number of
+        #  findings by product for SCP customers. See PA-3312
+        #  We should consider refactoring this logic into a shared function
+        #  in a future PR for metric and behavioral consistency.
+        #  An open question remains on whether we should be including the number
+        #  of ignored findings in this count.
+
+        findings_by_product: Dict[str, int] = Counter()
+        for r, f in matches_by_rule.items():
+            # NOTE: For parity with metrics.py, we are using the human-readable product name,
+            #  (i.e. code) and falling back to the internal json string (i.e. sast) if we
+            #  somehow drift out of sync with the product enum.
+            name = USER_FRIENDLY_PRODUCT_NAMES.get(r.product, r.product.to_json())
```

Comment:
optional: maybe `str(r.product.to_json())`, so there's no need to worry about the type of `name` being String or Any.

Alternatively for the next line, since there's only string conversion and no formatting, I'm wondering whether `str(name)` is clearer than `f"{name}"` or not.

Feel free to ignore and resolve. I feel like you always have a good reason to do things, so I'll assume I'm missing something.

Reply:
Here's the situation for a hopeful aha moment – 

On [line 354](https://github.com/semgrep/semgrep/pull/9682/files#diff-7c7cda58065ec85eebe55f7c9b79daed7e22b02b2ee553bb3379317a0ec4e505R354), name is a string in both the happy and fallback cases. 

Ideally, we would not need to call `.to_json()`on the product value if we had defined the `__str__` method for our dataclasses within our `atd` python code generator. Should we add this convenience feature to `atd`, usage of the f-string in the [next line](https://github.com/semgrep/semgrep/pull/9682/files#diff-7c7cda58065ec85eebe55f7c9b79daed7e22b02b2ee553bb3379317a0ec4e505R355) would be required vs a defensive programming technique for future-proofing and to please our little mypy checker.

Note that f-strings are both faster than `str` ([ref](https://stackoverflow.com/a/56587856)) and generally more flexible.

Reply:
> On line 354, name is a string in both the happy and fallback cases.

Hmm, now I'm wondering why mypy was complaining.

> Note that f-strings are both faster than str (ref) and generally more flexible.

Totally fine for this reason to use f-strings! I personally think `str` is clearer (aiming for clarity here), but I can be the minority who thinks that and it's a really minor point.